target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dts

```diff
@@ -0,0 +1,372 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "qcom-ipq8065.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Askey RT4230W";
+	compatible = "askey,rt4230w", "qcom,ipq8065";
+
+	memory@0 {
+		reg = <0x42000000 0x3e000000>;
+		device_type = "memory";
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+		rsvd@41200000 {
+			reg = <0x41200000 0x300000>;
+			no-map;
+		};
+	};
+
+	aliases {
+		serial0 = &gsbi4_serial;
+		mdio-gpio0 = &mdio0;
+		led-boot = &ledctrl3;
+		led-running = &ledctrl2;
```

Comment:
What about failsafe and upgrade indication?

Reply:
Is having boot and upgrade the same fine? Also is a red glow for failsafe fine?